tests/testJointcal_lsstSim.py

```diff
@@ -0,0 +1,134 @@
+# See COPYRIGHT file at the top of the source tree.
+
+from __future__ import division, absolute_import, print_function
+
+import unittest
+import os
+
+from lsst.afw import geom, coord
+import lsst.utils
+import lsst.pex.exceptions
+from lsst.daf import persistence
+from lsst.jointcal import jointcal
+import jointcalTestBase
+
+try:
+    data_dir = lsst.utils.getPackageDir('validation_data_jointcal')
+    os.environ['ASTROMETRY_NET_DATA_DIR'] = os.path.join(data_dir, 'twinkles1_and_index')
+except lsst.pex.exceptions.NotFoundError:
+    data_dir = None
+
+# We don't want the absolute astrometry to become significantly worse
+# than the single-epoch astrometry (about 0.040").
+absolute_error = 42e-3/jointcalTestBase.arcsec_per_radian
+# Set to True for a comparison plot and some diagnostic numbers.
+doPlot = False
+
+
+# for MemoryTestCase
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class FakeRef(object):
+    """
+    A mock data ref object, with minimal functionality.
+
+    This is needed to insert missing metadata while DM-5503 is dealt with.
+    """
+
+    def __init__(self, src, calexp_md, dataId):
+        self.src = src
+        self.calexp_md = calexp_md
+        # info about our "telescope"
+        # self.calexp_md.setDouble("AIRMASS", 1)
+        self.calexp_md.setDouble("MJD", 54321.)
+        # self.calexp_md.setDouble("EXPTIME", 30.)
+        self.calexp_md.setDouble("LST", 53.00914)
+        self.calexp_md.setDouble("RA2000", 53.00914)
+        self.calexp_md.setDouble("DEC2000", -27.43895)
+
+        self.dataId = dataId
+        self.dataId['ccd'] = 1
+        self.dataId['tract'] = 1
+
+    def get(self, name, immediate=True):
+        return getattr(self, name)
+
+    def put(self, value, name):
+        setattr(self, name, value)
+
+    def getButler(self):
+        class NamedThing(object):
+            def __init__(self, name):
+                self.name = name
+
+            def getName(self):
+                return self.name
+
+        return {'camera': NamedThing('monkeySim')}
```

Comment:
The above methods are all very nice, but I don't understand `getButler`.  In particular, shouldn't `getButler` return a `Butler`, not a `dict`?
I think you should rename `NamedThing` as `DummyCamera` because that's clearer about your intention.